golang/internal/helper/docker/network.go

```diff
@@ -0,0 +1,66 @@
+package docker
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/client"
+	"github.com/rs/zerolog/log"
+)
+
+func GetAllNetworks() ([]types.NetworkResource, error) {
+	ctx := context.Background()
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		panic(err)
+	}
+	return cli.NetworkList(ctx, types.NetworkListOptions{})
+}
+
+func CreateNetwork(ctx context.Context, name, driver string) error {
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		panic(err)
+	}
+
+	filter := filters.NewArgs()
+	filter.Add("name", name)
+
+	networkListOption := types.NetworkListOptions{
+		Filters: filter,
+	}
+
+	networks, err := cli.NetworkList(ctx, networkListOption)
+	if err != nil {
+		return fmt.Errorf("error list existing networks: %w", err)
+	}
+
+	if len(networks) > 0 {
+		log.Error().Msg("Provided network name: " + name + " exists. Skip to create new network.")
+		return nil
+	}
+
+	networkCreateOptions := types.NetworkCreate{
+		CheckDuplicate: true,
+		Driver:         driver,
+	}
+
+	_, err = cli.NetworkCreate(ctx, name, networkCreateOptions)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func DeleteNetworkByID(ctx context.Context, networkID string) error {
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		panic(err)
+	}
+
+	return cli.NetworkRemove(ctx, networkID)
```

Comment:
Don't we need to check errors after removing networks?

Reply:
We do need it, but in different place, where this func is called. (and as you might rightfully assume: there was none :D )

Reply:
Also fixed.